### ISSUE TITLE
How to avoid that Parallels mount my home folder in guest?

### ISSUE THREAD

**Issue by workoutweb** [Parallels/vagrant-parallels#171]
I have removed a lot of shared folders with this lines
`````

config.vm.provider "parallels" do |v|
	  v.customize ["set", :id, "--smart-mount", "off"]
	  v.customize ["set", :id, "--shared-profile", "off"]
      v.customize ["set", :id, "--shared-cloud", "off"]	

end
`````
but Home folders is already there
````
Home on /media/psf/Home type prl_fs (rw,nosuid,nodev,sync,noatime,share)
`````

**Comment by legal90** [comment]
Sorry, it is not possible to do this via Vagrant, because there is not such option in prlctl. :(
So, you can set in via GUI in the VM settings:
"Actions" -> "Configure..." -> "Options" -> "Sharing" -> "Share Folders" set to *None*
Custom shared folders, defined by Vagrant, are not affected by this setting and will be available in the guest OS.

P.s.
Could you please explain your use case? Is this undesired Home sharing critical for you?

**Comment by workoutweb** [comment]
Hi, Now I'm using the GUI to exclude the home from my project, but this settings is lost when I destroy the VM.

I think that have the home folder mounted by default is a big secure risk, because if I forget to disable it, I expose the home folder to an environment that should have some critical bug.

An another risk is about the sharing feature (http://docs.vagrantup.com/v2/share/ssh.html). I have never used it yet, but I think that share an ssh with home mounted can give access at reserved document to colleges or friends.

I hope that Parallels will add this option in prlctl as soon as possible.

Thank you for your answer. I have appreciated it a lot ;)

**Comment by legal90** [comment]
Thank you for an explanation. I'll keep you in touch here and will let you know when it will be implemented in prlctl. 

So, I think this issue can be closed.

**Comment by posita** [comment]
I haven't tested this, but In the mean time, as a hack, one could use [emyl/vagrant-triggers](/emyl/vagrant-triggers) and a script to force `umount`ing the offending directories. First, create the following (e.g., as `umountpsf.sh` in the same directory as your `Vagrantfile`):

```sh
#!/usr/bin/env sh
echo "executing ${0} ..."
set -ex

if -d /media/psf ; then
    for i in /media/psf/* ; do
        umount "${i}"
        rmdir -v "${i}"
    done

    rmdir -v /media/psf
fi

exit
```

Then call it on `vagrant up`:

```ruby
# Vagrantfile
    ...
    config.vm.provider :parallels do |v, override|
        config.trigger.after :up do
            run "vagrant ssh -c /vagrant/umountpsf.sh"
        end
    end
    ...
```

But I agree, this should *not* be necessary. Auto-mounting by default is a ***huge*** security no-no. :angry: At the very ***least***, Parallels' [own boxes](https://atlas.hashicorp.com/parallels) should completely disable all sharing by default except for `/vagrant` to remain consistent with other providers (but to be overridden as desired in the `Vagrantfile`). All of them I've seen have sharing ***on*** by default. :disappointed:

**Comment by posita** [comment]
*UPDATE: It appears that vagrant-triggers is not needed. For some reason (at least with v9), the following is sufficient (`/media/psf` is not recreated on subsequent runs):*

```ruby
# Vagrantfile
    ...
    config.vm.provider :parallels do |v, override|
        override.vm.provision :shell, :inline => '( set -ex ; if [ -d /media/psf ] ; then for i in /media/psf/* ; do umount "${i}" && rmdir -v "${i}" ; done ; rmdir -v /media/psf ; fi ; exit )'
    end
    ...
```

*Can someone verify whether this approach works on v10?*

--

I haven't tested this, but In the mean time, as a hack, one could use [emyl/vagrant-triggers](/emyl/vagrant-triggers) and a script to force `umount`ing the offending directories. First, create the following (e.g., as `umountpsf.sh` in the same directory as your `Vagrantfile`):

```sh
#!/usr/bin/env sh
set -ex

if [ -d /media/psf ] ; then
    for i in /media/psf/* ; do
        umount "${i}"
        rmdir -v "${i}"
    done

    rmdir -v /media/psf
fi

exit
```

Then call it on `vagrant up`:

```ruby
# Vagrantfile
    ...
    config.vm.provider :parallels do |v, override|
        override.vm.provision :shell, :path => 'umountpsf.sh'
        override.trigger.after :up do
            run "vagrant ssh -c /vagrant/umountpsf.sh"
        end
    end
    ...
```

But I agree, this should *not* be necessary. Auto-mounting by default is a ***huge*** security no-no. :angry: At the very least, Parallels' [own boxes](https://atlas.hashicorp.com/parallels) should completely disable all sharing by default except for `/vagrant` to remain consistent with other providers (but to be overridden as desired in the `Vagrantfile`). All of them I've seen have sharing ***on*** by default. :disappointed:

**Comment by legal90** [comment]
@chrvadala I'm sorry for the late response. The CLI option for managing this setting was implemented in Parallels Desktop 11.0.0. This is how to disable "Home" folder sharing:
```
prlctl set <vm_uuid> --shf-host-defined off
```
In Vagrantfile it could be done this way:
```
config.vm.provider "parallels" do |v|
      v.customize ["set", :id, "--shf-host-defined", "off"]
end
```

I will add this option to the list of default settings for all VMs, in addition to #215
